numpy/_core/src/multiarray/nditer_api.c

```diff
@@ -2187,9 +2196,8 @@ npyiter_copy_to_buffers(NpyIter *iter, char **prev_dataptrs)
             NPY_IT_DBG_PRINT("    marking operand %d for buffer reuse\n", iop);
             NIT_OPITFLAGS(iter)[iop] |= NPY_OP_ITFLAG_BUF_REUSABLE;
         }
-        else if (prev_dataptrs == NULL || prev_dataptrs[iop] != dataptrs[iop] ||
-                 bufferdata->coreoffset) {
```

Comment:
Seemed not useful to check.  We just always unset the flag (even when it isn't set).